src/plugins/vis_type_vislib/public/vislib/response_handler.js

```diff
@@ -42,6 +42,7 @@ function tableResponseHandler(table, dimensions) {
           title: `${splitColumnFormatter.convert(splitValue)}: ${splitColumn.name}`,
           name: splitColumn.name,
           key: splitValue,
+          formattedKey: splitColumnFormatter.convert(splitValue),
```

Comment:
nit: we don't need to calculate formatted value twice. 

 ```
 table.rows.forEach((row, rowIndex) => {
      const splitValue = row[splitColumn.id];
      const formattedValue = splitColumnFormatter.convert(splitValue);

      if (!splitMap.hasOwnProperty(splitValue)) {
        splitMap[splitValue] = splitIndex++;
        const tableGroup = {
          $parent: converted,
          title: `${formattedValue}: ${splitColumn.name}`,
          name: splitColumn.name,
          key: splitValue,
          formattedKey: formattedValue,
          column: splitColumnIndex,
          row: rowIndex,
          table,
          tables: [],
        };
```

Reply:
done